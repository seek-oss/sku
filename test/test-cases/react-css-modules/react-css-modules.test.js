/* eslint-disable jest/expect-expect */
const path = require('path');
const dirContentsToObject = require('../../utils/dirContentsToObject');
const runSkuScriptInDir = require('../../utils/runSkuScriptInDir');
const waitForUrls = require('../../utils/waitForUrls');
const startAssetServer = require('../../utils/assetServer');
const { getAppSnapshot } = require('../../utils/appSnapshot');
const appDir = path.resolve(__dirname, 'app');
const distDir = path.resolve(appDir, 'dist');
const storybookDistDir = path.resolve(appDir, 'dist-storybook');

const getStorybookContent = async (url) => {
  const page = await browser.newPage();
  await page.goto(url, { waitUntil: 'networkidle2' });

  const firstStoryButton = await page.waitForSelector(
    '#storybook-explorer-menu button',
  );

  // Ensure default story is activated
  firstStoryButton.click();

  const iframeElement = await page.waitForSelector('#storybook-preview-iframe');

  const storybookFrame = await iframeElement.contentFrame();

  if (!storybookFrame) {
    console.log('Unable to find storybookFrame', storybookFrame);
    throw new Error('Unable to find iframe by id');
  }

  const element = await storybookFrame.waitForSelector(
    '[data-automation-text]',
  );

  return element.evaluate((e) => ({
    text: e.innerText,
    fontSize: window.getComputedStyle(e).getPropertyValue('font-size'),
  }));
};

describe('react-css-modules', () => {
  let closeAssetServer;

  beforeAll(async () => {
    await runSkuScriptInDir('build', appDir);
    closeAssetServer = await startAssetServer(4293, distDir);
  });

  afterAll(() => {
    closeAssetServer();
  });

  it('should create valid app', async () => {
    const app = await getAppSnapshot('http://localhost:4293');
    expect(app).toMatchSnapshot();
  });

  it('should generate the expected files', async () => {
    const files = await dirContentsToObject(distDir);
    expect(files).toMatchSnapshot();
  });

  it('should handle Less and css.js in tests', async () => {
    const { childProcess } = await runSkuScriptInDir('test', appDir);
    expect(childProcess.exitCode).toEqual(0);
  });

  describe('storybook', () => {
    const storybookUrl = 'http://localhost:8081';
    let server;

    beforeAll(async () => {
      server = await runSkuScriptInDir('storybook', appDir, ['--ci']);
      await waitForUrls(storybookUrl);
    });

    afterAll(async () => {
      await server.kill();
    });

    it('should start a storybook server', async () => {
      const { text, fontSize } = await getStorybookContent(storybookUrl);
      expect(text).toEqual('Storybook render');
      expect(fontSize).toEqual('32px');
    });
  });

  describe('build-storybook', () => {
    let closeStorybookServer;

    beforeAll(async () => {
      await runSkuScriptInDir('build-storybook', appDir);
      closeStorybookServer = await startAssetServer(4297, storybookDistDir);
    });

    afterAll(() => {
      closeStorybookServer();
    });

    it('should create valid storybook', async () => {
      const { text, fontSize } = await getStorybookContent(
        'http://localhost:4297',
      );

      expect(text).toEqual('Storybook render');
      expect(fontSize).toEqual('32px');
    });
  });
});
