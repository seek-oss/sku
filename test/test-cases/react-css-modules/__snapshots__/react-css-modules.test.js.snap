// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-css-modules should create valid app 1`] = `
SCRIPTS: Array [
  "/runtime-9729bb82c0a343375b97.js",
  "/702-ef15ca26d5c62a3fad08.js",
  "/main-7a2fef10e729a96687d1.js",
]
CSS: Array [
  "/main-21ba36bf1c8257f9aa56.css",
]
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      My Awesome Project
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="main"
          rel="stylesheet"
          href="styles[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[1]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[2]"
    >
  </head>
  <body>
    <div id="app">
      <div class="lE4bKpu">
        <div class="_KPLLp3"
             data-automation-text="true"
        >
          Initial render
        </div>
      </div>
    </div>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
    >
      []
    </script>
    <script id="__LOADABLE_REQUIRED_CHUNKS___ext"
            type="application/json"
    >
      {"namedChunks":[]}
    </script>
    <script async
            data-chunk="main"
            src="scripts[0]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[1]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[2]"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: 
         <div class="_KPLLp3"
              data-automation-text="true"
         >
-          Initial render
+          Updated render
         </div>
       </div>
     </div>

`;

exports[`react-css-modules should generate the expected files 1`] = `
Object {
  "702-ef15ca26d5c62a3fad08.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "index.html": SCRIPTS: Array [
    "/runtime-9729bb82c0a343375b97.js",
    "/702-ef15ca26d5c62a3fad08.js",
    "/main-7a2fef10e729a96687d1.js",
  ]
CSS: Array [
    "/main-21ba36bf1c8257f9aa56.css",
  ]
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      My Awesome Project
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="main"
          rel="stylesheet"
          href="styles[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[1]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[2]"
    >
  </head>
  <body>
    <div id="app">
      <div class="lE4bKpu">
        <div class="_KPLLp3"
             data-automation-text="true"
        >
          Initial render
        </div>
      </div>
    </div>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
    >
      []
    </script>
    <script id="__LOADABLE_REQUIRED_CHUNKS___ext"
            type="application/json"
    >
      {"namedChunks":[]}
    </script>
    <script async
            data-chunk="main"
            src="scripts[0]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[1]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[2]"
    >
    </script>
  </body>
</html>,
  "main-21ba36bf1c8257f9aa56.css": .lE4bKpu {
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}
.lE4bKpu ._KPLLp3 {
  font-size: 32px;
}
,
  "main-7a2fef10e729a96687d1.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "runtime-9729bb82c0a343375b97.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
}
`;
